core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -843,14 +846,16 @@ oppia.factory('explorationStatesService', [
     var saveStateProperty = function(stateName, backendName, newValue) {
       var oldValue = getStatePropertyMemento(stateName, backendName);
       var newBackendValue = angular.copy(newValue);
+      var oldBackendValue = angular.copy(oldValue);
 
       if (BACKEND_CONVERSIONS.hasOwnProperty(backendName)) {
         newBackendValue = convertToBackendRepresentation(newValue, backendName);
+        oldBackendValue = convertToBackendRepresentation(oldValue, backendName);
       }
 
       if (!angular.equals(oldValue, newValue)) {
         changeListService.editStateProperty(
-          stateName, backendName, newBackendValue, oldValue);
+          stateName, backendName, newBackendValue, oldBackendValue);
```

Comment:
Why did this not result in an error before? Just wondering if it means we have any corrupted data in production.